internal/filesystem/filesystem.go

```diff
@@ -191,7 +190,7 @@ func (fs *fsystem) ReadFile(name string) ([]byte, error) {
 
 func (fs *fsystem) ReadDir(name string) ([]os.FileInfo, error) {
 	memList, err := afero.ReadDir(fs.memFs, name)
-	if err != nil {
+	if err != nil && !os.IsNotExist(err) {
```

Comment:
So this was actually a bug which wasn't visible before because the `ReadDir` codepath was not really exercised anywhere properly.

`Filesystem.ReadDir(name)` should always return a combination of filenames/directories in both virtual FS and on-disk FS. Not having this condition would cause this to error out when a new file was opened in editor and its parent directory doesn't exist (yet) on the disk.